### DISCUSSION TITLE
Narrow down our Microsoft Search to only documents inside single sharepoint site

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#2794] [Q&A]
We have added a new PnP Search Result web part >> with these settings:-

![image](https://user-images.githubusercontent.com/79726511/220456064-205c825b-225d-4228-849d-2a02920eac91.png)

but how we can only define to show documents from a single sharepoint site?

**Comment by kasperbolarsen** [comment]
Drive items, isn't that One Drive?

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen i am new to the Microsoft Search, i tried Drive and i got files from sharepoint!! any explanation?

**Reply by kasperbolarsen** [reply]
Sorry, I still have a hard time remembering the Entity Types in Graph. DriveItem is correct :-)

It should work, at least this config returns documents from the sites in my tenant :-) . I often use the debug layout or Graph Explorer to inspect the Managed Properties available in the result set.

![image](https://user-images.githubusercontent.com/20593570/220462463-96cd048c-c309-489e-9ebe-d2c35665e04a.png)

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  ok thanks for the reply. but how i can narrow it to  single sharepoint site only? so i only get documents from a single site?

**Reply by kasperbolarsen** [reply]
Exactly the same way as when using SharePoint as the date source, using path :-)

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  could you provide the formula beside the Isdocument? thanks

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  I tried the Path but this will not return any results, as follow:-

![image](https://user-images.githubusercontent.com/79726511/220476091-148c19b9-a797-4c34-877c-3b347e72bb4d.png)


any advice?
Thanks

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  Appreciate if you can further help with the above?
Thanks

**Reply by wobba** [reply]
Path takes a full path, not relative path `path:https://contoso.sharepoint.com/sites/hr`

**Comment by nattomatto** [comment]
You can get a single site by getting the ID of the site and using the siteID filter with in the Query template:  
![image](https://user-images.githubusercontent.com/9138840/224813729-4a66403e-f28f-4b5e-82ea-3e8205134a1c.png)
